pkgs/data/themes/adi1090x-plymouth-themes/default.nix

```
@@ -0,0 +1,63 @@
+{
+  stdenv,
+  fetchurl,
+  lib,
+  unzip,
+  # To select only certain themes, pass `selected_themes` as a list of strings.
+  # reference ./shas.nix for available themes
+  selected_themes ? [],
+}: let
+  # both of these files are generated via ./update.sh
+  # borrowed from pkgs/data/fonts/nerdfonts
+  version = import ./version.nix;
+  themeShas = import ./shas.nix;
+  knownThemes = builtins.attrNames themeShas;
+  selectedThemes =
+    if (selected_themes == [])
+    then knownThemes
+    else let
+      unknown = lib.subtractLists knownThemes selected_themes;
+    in
+      if (unknown != [])
+      then throw "Unknown theme(s): ${lib.concatStringsSep " " unknown}"
+      else selected_themes;
+  srcs = lib.lists.forEach selectedThemes (
+    name: (fetchurl {
+      url = themeShas.${name}.url;
+      sha256 = themeShas.${name}.sha;
+    })
+  );
+in
+  stdenv.mkDerivation {
+    inherit version;
+    inherit srcs;
+    pname = "adi1090x-plymouth-themes";
+    nativeBuildInputs = [
+      unzip
+    ];
+    sourceRoot = ".";
+    unpackCmd = "tar xzf $curSrc";
+    buildPhase = ''
+      echo "selected themes are ${toString selectedThemes}"
+    '';
+    installPhase = ''
+      mkdir -p $out/share/plymouth/themes
+      for theme in ${toString selectedThemes}; do
+        mv $theme $out/share/plymouth/themes/$theme
+      done
+      find $out/share/plymouth/themes/ -name \*.plymouth -exec sed -i "s@\/usr\/@$out\/@" {} \;
+    '';
+    meta = with lib; {
```

```suggestion
  stdenv.mkDerivation {
    pname = "adi1090x-plymouth-themes";
    inherit version srcs;

    nativeBuildInputs = [
      unzip
    ];

    sourceRoot = ".";
    unpackCmd = "tar xzf $curSrc";

    installPhase = ''
      mkdir -p $out/share/plymouth/themes
      for theme in ${toString selectedThemes}; do
        mv $theme $out/share/plymouth/themes/$theme
      done
      find $out/share/plymouth/themes/ -name \*.plymouth -exec sed -i "s@\/usr\/@$out\/@" {} \;
    '';

    meta = with lib; {
```
lets make that a bit nicer